src/Model.php

```diff
@@ -122,6 +127,18 @@ public function createRelations(Relations $relations)
     {
     }
 
+    /**
+     * Create default relations to utilize (join)
+     *
+     * This model relations aren't real relations, where all model columns are also selected,
+     * rather they are only being used for utilizing (joining)
+     *
+     * @param Query $query
+     */
+    public function createDefaultUtilize(Query $query)
```

Comment:
Why not `prepareQuery` (my favorite) or `initQuery`? The current name implies something that cannot be controlled anyway. So I don't think we should try to. The docstring also has to change in this case.